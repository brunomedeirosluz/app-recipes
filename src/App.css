.meals {
  align-items: center;
  background: #0fa36b;
  color: #464646;
  display: flex;
  flex-direction: column;
  font-family: sans-serif;
  height: 100vh;
  justify-content: center;
  width: 100%;
}

.rocksGlass {
  animation: shake 1s infinite ease;
  max-height: 80vh;
}

.footer {
  background-color: #41197F;
  bottom: 0;
  box-sizing: border-box;
  display: flex;
  justify-content: space-between;
  padding: 8px;
  position: fixed;
  width: 100%;
}

.start-btn {
  bottom: 0px;
  display: flex;
  left: 50%;
  position: fixed;
  transform: translateX(-50%);
}

.recipe-card {
  border: 1px solid #ccc;
  box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
  box-sizing: border-box;
  display: inline-block;
  margin: 16px;
  padding: 16px;
  text-align: center;
  vertical-align: top;
  width: 200px;
}

.recipe-card img {
  border-radius: 8px;
  height: auto;
  max-width: 100%;
}

.recipe-card p {
  font-weight: 700;
  margin-top: 7px;
}

/*
#iceLeft {
  animation: clink .8s infinite ease;
  opacity: 0.6;
  transform-origin: 36px 66px;
}

#iceRight {
  animation: clink .8s .2s infinite ease;
  opacity: 0.6;
  transform-origin: 60px 78px;
}
*/

.meals > .logo {
  animation: shake 1.5s infinite ease;
  color: aliceblue;
  font-family: Montserrat , sans-serif;
  font-size: 3em;
  font-weight: 900;
}

@keyframes shake {

  0% {
    transform: rotate(5deg);
  }

  50% {
    transform: rotate(-5deg);
  }

  100% {
    transform: rotate(5deg);
  }
}

@keyframes clink {

  0% {
    transform: rotate(-10deg);
  }

  50% {
    transform: rotate(20deg);
  }

  100% {
    transform: rotate(-10deg);
  }
}
