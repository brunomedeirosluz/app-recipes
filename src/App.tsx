import { Routes, Route } from 'react-router-dom';
import './App.css';
import Login from './pages/Login';
import Meals from './pages/Meals';
import Drinks from './pages/Drinks';
import MealsDetails from './pages/MealsDetails';
import DrinksDetails from './pages/DrinksDetails';
import MealProgress from './pages/MealsProgress';
import DrinksProgress from './pages/DrinksProgress';
import Profile from './pages/Profile';
import DoneRecipes from './pages/Done-recipes';
import FavoriteRecipes from './pages/Favorite-recipes';

export default function App() {
  return (
    <Routes>
      <Route path="/" element={ <Login /> } />
      <Route path="/meals/:id/in-progress" element={ <MealProgress /> } />
      <Route path="/meals/:id" element={ <MealsDetails /> } />
      <Route path="/meals" element={ <Meals /> } />
      <Route path="/drinks/:id/in-progress" element={ <DrinksProgress /> } />
      <Route path="/drinks/:id" element={ <DrinksDetails /> } />
      <Route path="/drinks" element={ <Drinks /> } />
      <Route path="/profile" element={ <Profile /> } />
      <Route path="/done-recipes" element={ <DoneRecipes /> } />
      <Route path="/favorite-recipes" element={ <FavoriteRecipes /> } />
    </Routes>
  );
}
