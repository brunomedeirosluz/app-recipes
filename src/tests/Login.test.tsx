import React from 'react';
import { fireEvent, screen } from '@testing-library/react';
import App from '../App';
import { renderWithRouter } from '../helps/renderWithRouter';

test.only('O botão deve estar desativado se o email e a senha forem inválido', async () => {
  const { user } = renderWithRouter(
    <App />,
  );

  const emailInput = screen.getByTestId('email-input');
  const passwordInput = screen.getByTestId('password-input');
  const loginSubmitButton = screen.getByTestId('login-submit-btn');

  expect(loginSubmitButton).toBeDisabled();

  await fireEvent.change(emailInput, { target: { value: 'email@email.com' } });
  await fireEvent.change(passwordInput, { target: { value: 'senhaSegura' } });

  expect(loginSubmitButton).toBeInTheDocument();

  await user.click(loginSubmitButton);

  const meals = await screen.getByRole('heading', { name: /meals page/i });
  expect(meals).toBeInTheDocument();
});

test('O botão deve estar ativado se o email e a senha forem válidos', () => {
  renderWithRouter(
    <App />,
  );

  const emailInput = screen.getByTestId('email-input');
  const passwordInput = screen.getByTestId('password-input');
  const loginSubmitButton = screen.getByTestId('login-submit-btn');

  fireEvent.change(emailInput, { target: { value: 'email@exemplo.com' } });
  fireEvent.change(passwordInput, { target: { value: 'senhaSegura' } });

  expect(loginSubmitButton).toBeEnabled();
});
