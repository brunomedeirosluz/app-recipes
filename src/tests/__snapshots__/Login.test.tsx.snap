// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Teste da tela de login > Verifica a cobertura de 45% da tela de Login 1`] = `
{
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "logTestingPlaygroundURL": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
}
`;

exports[`Verifica a cobertura de 45% da tela de Login 1`] = `
<div>
  <div>
    <form>
      <input
        data-testid="email-input"
        name="email"
        placeholder="E-Mail"
        type="email"
        value=""
      />
      <input
        data-testid="password-input"
        name="password"
        placeholder="Password"
        type="password"
        value=""
      />
      <button
        data-testid="login-submit-btn"
        disabled=""
        type="submit"
      >
        Enter
      </button>
    </form>
  </div>
</div>
`;
